### PR TITLE
Bump @nextcloud/vue from 7.5.0 to 7.7.0 and add node-polyfill-webpack-plugin

### PR DESCRIPTION
**Pull request by Pytal**
## Summary

Bump `@nextcloud/vue` from https://github.com/nextcloud/nextcloud-vue/releases/tag/v7.5.0 to https://github.com/nextcloud/nextcloud-vue/releases/tag/v7.7.0

Add https://github.com/Richienb/node-polyfill-webpack-plugin to fix the `process is not defined` and resulting `OC is not defined` errors

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [x] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Review by szaimen — APPROVED** [review]
🐘

**Comment by artonge** [comment]
Shouldn't the process thing be solved on the `@nextcloud/files` side ?

File: webpack.common.js [hunk 1]
```diff
@@ -161,6 +164,9 @@ module.exports = {
 
 			// https://github.com/webpack/changelog-v5/issues/10
 			Buffer: ['buffer', 'Buffer'],
+
+			// Provide process module shim for dependencies which import it
+			process: 'process/browser',
```
Comment (juliushaertl) [inline webpack.common.js:169]: We could think about using https://www.npmjs.com/package/node-polyfill-webpack-plugin

**Review by juliusknorr — APPROVED** [review]

**Comment by nickvergessen** [comment]
> FAIL apps/dav/src/views/CalDavSettings.spec.js

**Comment by Pytal** [comment]
> Shouldn't the process thing be solved on the `@nextcloud/files` side ? I never really played with modules, so I have no idea.

`@nextcloud/files` (https://github.com/nextcloud/nextcloud-files/blob/94bfe0e1a093dcb8ed6b0710e7bdd8214ba16427/lib/files/node.ts#L22) seems to have an implicit dependency on running within a https://github.com/nextcloud/server runtime environment as it neither has a `path` dependency in [package.json](https://github.com/nextcloud/nextcloud-files/blob/94bfe0e1a093dcb8ed6b0710e7bdd8214ba16427/package.json) nor a polyfill for it in https://github.com/nextcloud/nextcloud-files/blob/94bfe0e1a093dcb8ed6b0710e7bdd8214ba16427/rollup.config.js while the server [package.json](https://github.com/nextcloud/server/blob/00a7478d8df2007697d5caf2e0cb1c298afbb3fd/package.json#L89) does

From the `process is not defined` error stack trace we see that the `path` import is resolved to https://github.com/jinder/path which is also used in https://github.com/nextcloud/server https://github.com/nextcloud/server/blob/00a7478d8df2007697d5caf2e0cb1c298afbb3fd/apps/files/src/views/TemplatePicker.vue#L67 https://github.com/nextcloud/server/blob/00a7478d8df2007697d5caf2e0cb1c298afbb3fd/apps/files_sharing/src/components/SelectShareFolderDialogue.vue#L43

For these reasons let's handle the dependencies and polyfills here for now, while explicit dependencies and polyfills may be added to https://github.com/nextcloud/nextcloud-files separately if needed

**Pull request merged**